### PR TITLE
Mrn facemask dev

### PR DESCRIPTION
**Pull request by mrn-mln**
#68 
This codebase is useful for training and inferencing face-mask.
I will add a license and also docstring to this branch.

### CODE REVIEW

File: applications/facemask/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* X86 node
+
+#### Software
+
+* You should have [Docker](https://docs.docker.com/get-docker/) on your device.
+
+### Install
+Make sure you have the prerequisites and then clone this repository to your local system by running this command:
+```
+git clone https://github.com/neuralet/face-mask.git
```
Comment (JsonSadler) [inline applications/facemask/README.md]: This does not seem right, please correct the repo address and path

File: applications/facemask/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+cd facemask/
+```
+## Usage
+The application has a module for trainig Face-Mask classifier and three inference mode:
+1- Inference on app mode
```
Comment (JsonSadler) [inline applications/facemask/README.md]: What is app mode?

File: applications/facemask/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+### Run on x86
+```
+# 1) Build Docker image
+docker build -f x86.Dockerfile -t "neuralet/face-mask:latest-x86" .
+
+# 2) Run Docker container:
+docker run  -it --gpus all -p HOST_PORT:8000 -v "$PWD/../../":/repo/ -it neuralet/face-mask:latest-x86
+
+# 3) Run main application python script
+python inference_main_app.py --config configs/config.json
```
Comment (JsonSadler) [inline applications/facemask/README.md]: just got: `failed to load video  data/video/sample2.mov`, any script for downloading the sample? You have also add it to the readme.

Comment (JsonSadler) [reply applications/facemask/README.md]: also it is good to mention that the user will be inside the docker after running the previous command, and has to enter the third step inside the docker.

File: applications/facemask/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+* NOTE: There is a config file at `configs/` directory for customizing the parameters of the model and the application. Please set the parameters if you plan to have a customized setting  
+
+### Run on x86
+```
+# 1) Build Docker image
+docker build -f x86.Dockerfile -t "neuralet/face-mask:latest-x86" .
+
+# 2) Run Docker container:
+docker run  -it --gpus all -p HOST_PORT:8000 -v "$PWD/../../":/repo/ -it neuralet/face-mask:latest-x86
```
Comment (JsonSadler) [inline applications/facemask/README.md:43]: please explain the HOST_PORT , and why is that required? Does this PR uses a web-gui?

Comment (JsonSadler) [reply applications/facemask/README.md:43]: By the way, I see that you have passed `--gpus` to your docker run command, if GPU is required, please rename this to something like x86-gpu or at least clear this in the readme that the device must have a working Nvidia GPU and Cuda (with required version). If it is not required, please remove `--gpus all`

File: applications/facemask/configs/config_edgetpu.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  "detector": {
+    "name": "posenet",
+    "input_size": [1281,721]
+  },
+  "app": {
+    "device": "EdgeTPU",
+    "video_path": "data/sample2.mov",
+    "resolution": [320, 320],
+    "host": "0.0.0.0",
+    "port": "8000"
```
Comment (JsonSadler) [inline applications/facemask/configs/config_edgetpu.json:43]: please use the same port in all devices.

File: applications/facemask/configs/config.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (JsonSadler) [inline applications/facemask/configs/config.json]: better to be config-x86 if this is used for x86 run.

File: applications/facemask/configs/config.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  "detector": {
+    "name": "openpifpaf",
+    "input_size":  [1281,721]
+  },
+  "app": {
+    "device": "x86",
+    "video_path": "data/video/sample2.mov",
+    "resolution": [320, 320],
+    "host": "0.0.0.0",
+    "port": "6006"
```
Comment (JsonSadler) [inline applications/facemask/configs/config.json]: why 6006? readme says 8000, what am I missing?

File: applications/facemask/README.md (hunk 4 above)
Comment (JsonSadler) [reply applications/facemask/README.md:43]: I ran this, it uses CPU but not GPU, is it correct?

File: applications/facemask/x86.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
+FROM pytorch/pytorch:1.5-cuda10.1-cudnn7-runtime
+
+VOLUME  /repo
+WORKDIR /repo/applications/facemask
+
+RUN apt-get update && apt-get install -y pkg-config libsm6 libxext6 libxrender-dev libgl1-mesa-glx
+
+RUN pip install --upgrade pip setuptools==41.0.0 && pip install opencv-python wget flask scipy image tensorflow-gpu==1.15.0 openpifpaf keras==2.3.0 matplotlib scikit-learn
+
+EXPOSE 8000
```
Comment (JsonSadler) [inline applications/facemask/x86.Dockerfile]: remove the export please, as running instructions invoke `-p ...` and the port is configurable in the configs, so it is both redundant and incontinent.

File: applications/facemask/README.md (hunk 2 above)
Comment (mrn-mln) [reply applications/facemask/README.md]: Inference on app mode (Using WebGUI to show the result)

File: applications/facemask/README.md (hunk 3 above)
Comment (mrn-mln) [reply applications/facemask/README.md]: added
3) Run main application python script inside the docker

File: applications/facemask/README.md (hunk 4 above)
Comment (mrn-mln) [reply applications/facemask/README.md:43]: It should use GPU automatically but it seems something is wrong with TensorFlow inside the docker. I will try to fix it.

**Review by mhejrati — APPROVED** [review]
Great work, thank you.
